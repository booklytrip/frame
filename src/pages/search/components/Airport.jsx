/**
 * @flow
 */

import React from 'react';
import styled from 'styled-components';
import { Search as BaseSearch, Flag, Icon } from 'semantic-ui-react';
import { toLower } from 'lodash';

type AirportOptionProps = {
    label: string,
    value: string,
    cityCode: string,
    countryCode: string,
    iata: string,
    name: string,
    shift: boolean,
};

type AirportProps = {
    loading: boolean,
    results: Array<AirportOptionProps>,
    text: string,
    placeholder: string,
    onSearchChange: Function,
    onResultSelect: Function,
    onSwap: Function,
    onBlur: Function,
};

const ShiftIcon = styled(Icon)`
    margin-left: 1em !important;
`;

const Search = styled(BaseSearch)`
    input {
        height: 56px !important;
        border-radius: 4px !important;
    }
`;

const SwapIcon = styled.div`
    cursor: pointer;
    position: absolute;
    z-index: 20;
    right: 14px;
    top: 18px;
    width: 18px;
    height: 18px;

    &:hover {
        &:before {
            transform: translateX(-2px);
        }
        &:after {
            transform: translateX(2px);
        }
    }

    &:before {
        bottom: 0;
        left: 0;
        color: #2185D0;
        font-family: Icons;
        content: "\uF177";
        display: block;
        position: absolute;
        transition: transform .2s ease-in-out,-o-transform .2s ease-in-out;
    }

    &:after {
        color: #2185D0;
        font-family: Icons;
        content: "\uF178";
        display: block;
        position: absolute;
        transition: transform .2s ease-in-out,-o-transform .2s ease-in-out;
        top: 8px;
        left: 2px;
    }
`;

const IataCode = styled.div`
    display: block;
    position: absolute;
    top: 0;
    right: 0px;
    font-size: 0.8em;
    line-height: 18px;
    color: #999;
`;

const AirportOptionContent = styled.div`
    position: relative;
    padding-right: 30px;
`;

const AirportOption = ({
    label,
    shift,
    countryCode,
    cityCode,
    iata,
}: AirportOptionProps) => (
    <AirportOptionContent>
        {shift
            ? <ShiftIcon name="chevron right" />
            : <Flag name={toLower(countryCode)} />}
        {label}
        <IataCode>{iata || cityCode}</IataCode>
    </AirportOptionContent>
);

const Airport = ({
    loading,
    text,
    placeholder,
    results,
    onSearchChange,
    onResultSelect,
    onSwap,
    onBlur,
}: AirportProps) => (
    <Search
        loading={loading}
        value={text}
        placeholder={placeholder}
        icon={loading || !onSwap ? <Icon /> : <SwapIcon onClick={onSwap} />}
        results={results}
        resultRenderer={AirportOption}
        onSearchChange={onSearchChange}
        onResultSelect={onResultSelect}
        onBlur={onBlur}
        selectFirstResult
    />
);

export default Airport;
