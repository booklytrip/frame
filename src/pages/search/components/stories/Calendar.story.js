import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import Calendar from '../Calendar';

storiesOf('search/Calendar', module)
    .addDecorator(centered)
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .add('default', () => {
        return <Calendar placeholder="Select" />;
    })
    .add('clearable', () => {
        return (
            <Calendar
                placeholder="Select"
                onClear={action('onClear')}
                onChange={action('onChange')}
                value="2017-04-11"
                clearable
            />
        );
    });
