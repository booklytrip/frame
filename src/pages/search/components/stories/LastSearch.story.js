// import React from 'react';
// import { storiesOf } from '@storybook/react';
// import centered from '@kadira/react-storybook-decorator-centered';

// import { ReduxProvider } from '../../../../providers';

// import LastSearch from '../LastSearch.jsx';

// import { lastSearchQuries } from './fixtures';

// storiesOf('search/LastSearch', module)
//     .addDecorator(story =>
//         <ReduxProvider>
//             {story()}
//         </ReduxProvider>,
//     )
//     .addDecorator(centered)
//     .add('one item', () =>
//         <LastSearch queries={lastSearchQuries.slice(0, 1)} />,
//     )
//     .add('few items', () => <LastSearch queries={lastSearchQuries} />);
