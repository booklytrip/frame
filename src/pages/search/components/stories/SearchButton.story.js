import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import { Container } from 'semantic-ui-react';
import SearchButton from '../SearchButton';

storiesOf('search/SearchButton', module)
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(story =>
        <Container>
            {story()}
        </Container>,
    )
    .addDecorator(centered)
    .add('default', () => <SearchButton onClick={action('onClick')} />)
    .add('loading', () => <SearchButton onClick={action('onClick')} loading />);
