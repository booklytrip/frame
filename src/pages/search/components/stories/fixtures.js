export const airports = [
    {
        label: 'Bertrix',
        name: 'Bertrix',
        countryCode: 'BE',
        cityCode: 'BE',
    },
    {
        iata: 'TMT',
        label: 'Trombetas',
        name: 'Trombetas',
        countryCode: 'BR',
        cityCide: 'BR',
    },
    {
        iata: 'AUA',
        label: 'Reina Beatrix Intl',
        name: 'Reina Beatrix Intl',
        countryCode: 'AW',
        cityCide: 'AW',
    },
    {
        iata: 'QQS',
        label: 'St Pancras Railway Station',
        name: 'St Pancras Railway Station',
        countryCode: 'GB',
        cityCide: 'LON',
    },
];

export const oneWaySearchQuery = {
    type: 'ONE_WAY',
    departureAirport: {
        code: 'RIX',
        city: {
            name: 'Riga',
        },
    },
    arrivalAirport: {
        code: 'STN',
        city: {
            name: 'London',
        },
    },
    departureDate: '2016-01-09 15:45:03',
    returnDate: '',
    adultPassengers: 1,
    childPassengers: '',
    infantPassengers: '',
};

export const roundTripSearchQuery = {
    type: 'ROUND_TRIP',
    departureAirport: {
        code: 'VNO',
        city: {
            name: 'Vilnius',
        },
    },
    arrivalAirport: {
        code: 'STN',
        city: {
            name: 'London',
        },
    },
    departureDate: '2016-01-04 15:45:03',
    returnDate: '2016-01-28 13:05:00',
    adultPassengers: 1,
    childPassengers: 2,
    infantPassengers: '',
};

export const lastSearchQuries = [oneWaySearchQuery, roundTripSearchQuery];
