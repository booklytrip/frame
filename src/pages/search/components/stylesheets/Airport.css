.airport .Select-control {
    font-size: 15px;
    height: 54px;
    line-height: 54px;
}

.airport .Select-placeholder,
.airport .Select--single > .Select-control .Select-value {
    line-height: inherit;
    padding: 0px 36px 0px 18px;
}

.airport .Select-input {
    height: inherit;
    padding: 0px 36px 0px 18px;
    display: block !important;
}

.airport.swap .Select-value {
    padding-right: 40px !important;
}

.airport .Select-arrow-zone {
    display: none;
}

.airport .Select-clear-zone {
    right: 14px;
}

.airport .swap {
    cursor: pointer;
    position: absolute;
    z-index: 20;
    right: 14px;
    top: 18px;
    width: 18px;
    height: 18px;
}

.airport .swap:hover:before {
    transform: translateX(-2px);
}

.airport .swap:hover:after {
    transform: translateX(2px);
}

.airport .swap:before {
    bottom: 0;
    left: 0;
    color: #2185D0;
    font-family: 'Icons';
    content: "\f177";
    display: block;
    position: absolute;
    -o-transition: -o-transform .2s ease-in-out;
    transition: transform .2s ease-in-out,-o-transform .2s ease-in-out;
}

.airport .swap:after {
    color: #2185D0;
    font-family: 'Icons';
    content: "\f178";
    display: block;
    position: absolute;
    -o-transition: -o-transform .2s ease-in-out;
    transition: transform .2s ease-in-out,-o-transform .2s ease-in-out;
    top: 8px;
    left: 2px;
}
