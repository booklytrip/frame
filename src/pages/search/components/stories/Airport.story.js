import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import Airport from '../Airport.jsx';

import { airports as airportsList } from './fixtures.js';

storiesOf('search/Airport', module)
    .addDecorator(centered)
    .add('default', () => <Airport placeholder="Departure place" />)
    .add('loading', () => <Airport placeholder="Departure place" loading />)
    .add('with results', () =>
        <Airport
            placeholder="Departure place"
            results={airportsList}
            onResultSelect={action('onResultSelect')}
        />,
    )
    .add('with grouped results', () =>
        <Airport
            placeholder="Departure place"
            results={airportsList.map((airport, idx) => ({
                ...airport,
                shift: idx > 0,
            }))}
            onResultSelect={action('onResultSelect')}
        />,
    )
    .add('with swap icon', () =>
        <Airport placeholder="Departure place" onSwap={action('onSwap')} />,
    );
