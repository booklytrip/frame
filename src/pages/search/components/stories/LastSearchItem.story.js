import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@kadira/react-storybook-decorator-centered';

import { oneWaySearchQuery } from './fixtures';

import LastSearchItem from '../LastSearchItem.jsx';

storiesOf('search/LastSearchItem', module)
    .addDecorator(centered)
    .add('default', () =>
        <LastSearchItem
            departureAirport={oneWaySearchQuery.departureAirport}
            arrivalAirport={oneWaySearchQuery.arrivalAirport}
        />,
    )
    .add('loading', () =>
        <LastSearchItem
            departureAirport={oneWaySearchQuery.departureAirport}
            arrivalAirport={oneWaySearchQuery.arrivalAirport}
            loading
        />,
    );
