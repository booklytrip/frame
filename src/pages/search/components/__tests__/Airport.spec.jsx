import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Airport from '../Airport';

import { airports } from '../stories/fixtures';

describe('Airport', () => {
    it('should render without crashing', () => {
        mount(<Airport />);
    });

    describe('Snapshots', () => {
        it('should render correctly', () => {
            const rendered = renderer.create(<Airport />).toJSON();
            expect(rendered).toMatchSnapshot();
        });

        it('should render results', () => {
            const rendered = renderer
                .create(<Airport results={airports} />)
                .toJSON();
            expect(rendered).toMatchSnapshot();
        });

        it('should render grouped results', () => {
            const grouped = airports.map((airport, idx) => ({
                ...airport,
                shift: idx > 0,
            }));

            const rendered = renderer
                .create(<Airport results={grouped} />)
                .toJSON();
            expect(rendered).toMatchSnapshot();
        });
    });
});
