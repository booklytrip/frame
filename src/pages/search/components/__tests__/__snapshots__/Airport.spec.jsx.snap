// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport Snapshots should render correctly 1`] = `
<div
  className="ui search gFTPLJ"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="ui icon input"
  >
    <input
      autoComplete="off"
      className="prompt"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      tabIndex="0"
      type="text"
      value=""
    />
    <i
      aria-hidden="true"
      className="icon"
    />
  </div>
  <div
    className="results transition"
  >
    <div
      className="message empty"
    >
      <div
        className="header"
      >
        No results found.
      </div>
    </div>
  </div>
</div>
`;

exports[`Airport Snapshots should render grouped results 1`] = `
<div
  className="ui search gFTPLJ"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="ui icon input"
  >
    <input
      autoComplete="off"
      className="prompt"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      tabIndex="0"
      type="text"
      value=""
    />
    <i
      aria-hidden="true"
      className="icon"
    />
  </div>
  <div
    className="results transition"
  >
    <div
      cityCode="BE"
      className="active result"
      countryCode="BE"
      label="Bertrix"
      name="Bertrix"
      onClick={[Function]}
      shift={false}
    >
      <div
        className="bTXEqs"
      >
        <i
          className="be flag"
        />
        Bertrix
        <div
          className="efSUMV"
        >
          BE
        </div>
      </div>
    </div>
    <div
      cityCide="BR"
      className="result"
      countryCode="BR"
      iata="TMT"
      label="Trombetas"
      name="Trombetas"
      onClick={[Function]}
      shift={true}
    >
      <div
        className="bTXEqs"
      >
        <i
          aria-hidden="true"
          className="chevron right icon kFUbcp"
        />
        Trombetas
        <div
          className="efSUMV"
        >
          TMT
        </div>
      </div>
    </div>
    <div
      cityCide="AW"
      className="result"
      countryCode="AW"
      iata="AUA"
      label="Reina Beatrix Intl"
      name="Reina Beatrix Intl"
      onClick={[Function]}
      shift={true}
    >
      <div
        className="bTXEqs"
      >
        <i
          aria-hidden="true"
          className="chevron right icon kFUbcp"
        />
        Reina Beatrix Intl
        <div
          className="efSUMV"
        >
          AUA
        </div>
      </div>
    </div>
    <div
      cityCide="LON"
      className="result"
      countryCode="GB"
      iata="QQS"
      label="St Pancras Railway Station"
      name="St Pancras Railway Station"
      onClick={[Function]}
      shift={true}
    >
      <div
        className="bTXEqs"
      >
        <i
          aria-hidden="true"
          className="chevron right icon kFUbcp"
        />
        St Pancras Railway Station
        <div
          className="efSUMV"
        >
          QQS
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Airport Snapshots should render results 1`] = `
<div
  className="ui search gFTPLJ"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="ui icon input"
  >
    <input
      autoComplete="off"
      className="prompt"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      tabIndex="0"
      type="text"
      value=""
    />
    <i
      aria-hidden="true"
      className="icon"
    />
  </div>
  <div
    className="results transition"
  >
    <div
      cityCode="BE"
      className="active result"
      countryCode="BE"
      label="Bertrix"
      name="Bertrix"
      onClick={[Function]}
    >
      <div
        className="bTXEqs"
      >
        <i
          className="be flag"
        />
        Bertrix
        <div
          className="efSUMV"
        >
          BE
        </div>
      </div>
    </div>
    <div
      cityCide="BR"
      className="result"
      countryCode="BR"
      iata="TMT"
      label="Trombetas"
      name="Trombetas"
      onClick={[Function]}
    >
      <div
        className="bTXEqs"
      >
        <i
          className="br flag"
        />
        Trombetas
        <div
          className="efSUMV"
        >
          TMT
        </div>
      </div>
    </div>
    <div
      cityCide="AW"
      className="result"
      countryCode="AW"
      iata="AUA"
      label="Reina Beatrix Intl"
      name="Reina Beatrix Intl"
      onClick={[Function]}
    >
      <div
        className="bTXEqs"
      >
        <i
          className="aw flag"
        />
        Reina Beatrix Intl
        <div
          className="efSUMV"
        >
          AUA
        </div>
      </div>
    </div>
    <div
      cityCide="LON"
      className="result"
      countryCode="GB"
      iata="QQS"
      label="St Pancras Railway Station"
      name="St Pancras Railway Station"
      onClick={[Function]}
    >
      <div
        className="bTXEqs"
      >
        <i
          className="gb flag"
        />
        St Pancras Railway Station
        <div
          className="efSUMV"
        >
          QQS
        </div>
      </div>
    </div>
  </div>
</div>
`;
